### PR TITLE
cleanup package.json and move all dependencies to peerDependencies

### PR DESCRIPTION
**Pull request by fschade**
before this pr the package.json contained a lot of unnecessary dependencies which where hard linked.
This pr moves them to the peerDependencies and let the consuming application decide which patch or minor version to use.

We also had a lot of unused dependencies before.

Because of the way dependency management works this pr introduces breaking changes and a major release is required.

### CODE REVIEW

**Review by LukasHirt — APPROVED** [review]
LGTM 👍

**Review by JammingBen — COMMENTED** [review]
LGTM, although pipeline disagrees :D Run `yarn test -u` and see what's changed?

File: package.json [hunk 1]
```diff
@@ -2,13 +2,25 @@
...
   "description": "ownCloud Design System is based on VueDesign Systems and is used to design ownCloud UI components",
+  "keywords": [
+    "vue design system",
+    "design system",
+    "vue.js",
+    "vue",
+    "design",
+    "system"
+  ],
+  "homepage": "https://owncloud.github.io/owncloud-design-system/",
```
Comment (pascalwengerter) [inline package.json:13]: Could be owncloud.design for a nice URL :)

**Review by pascalwengerter — CHANGES_REQUESTED** [review]
Missing a changelog item and one suggestion to improve the homepage URL, otherwise seems good to go!

File: package.json (hunk 1 above)
Comment (fschade) [reply package.json]: done, thanks.

**Review by pascalwengerter — APPROVED** [review]

**Pull request merged**